src/new/wallet.js

```diff
@@ -1944,50 +2059,57 @@ class HathorWallet extends EventEmitter {
+   * @param {MeltTokensOptions} [options] Options parameters
+   *
+   * @return {Promise<SendTransaction>}
+   *
+   * @memberof HathorWallet
+   * @inner
+   * */
+  async meltTokensSendTransaction(tokenUid, amount, options = {}) {
+    const transaction = await this.prepareMeltTokensData(tokenUid, amount, options);
+    return new SendTransaction({ storage: this.storage, transaction });
+  }
+
   /**
    * Melt tokens
    *
    * @param {string} tokenUid UID of the token to melt
    * @param {OutputValueType} amount Quantity to melt
-   * @param [options] Options parameters
-   * @param {string} [options.address]: address of the HTR deposit back
-   * @param {string} [options.changeAddress] address of the change output
-   * @param {boolean} [options.createAnotherMelt] boolean to create another melt authority or not
-   *                                              for the wallet
-   * @param {string} [options.meltAuthorityAddress] address to send the new melt authority created
-   * @param {boolean} [options.allowExternalMeltAuthorityAddress=false] allow the melt authority address
-   *                                                                    to be from another wallet
-   * @param {boolean} [options.startMiningTx=true] boolean to trigger start mining (default true)
-   * @param {boolean} [options.unshiftData=false] boolean to unshift the data script output
-   * @param {string[]|null} [options.data=null] list of data strings using utf8 encoding to add each as a data script output
-   * @param {string} [options.pinCode] pin to decrypt xpriv information.
-   *                                   Optional but required if not set in this
+   * @param {MeltTokensOptions} [options] Options parameters
    *
-   * @return {Promise<BaseTransactionResponse>} Promise that resolves with transaction object
-   *                                            if it succeeds or with error message if it fails
+   * @return {Promise<Transaction>}
    *
    * @memberof HathorWallet
    * @inner
    * */
   async meltTokens(tokenUid, amount, options = {}) {
-    const tx = await this.prepareMeltTokensData(tokenUid, amount, options);
-    return this.handleSendPreparedTransaction(tx);
+    const sendTx = await this.meltTokensSendTransaction(tokenUid, amount, options);
+    return sendTx.runFromMining();
   }
 
+  /**
+   * @typedef {Object} DelegateAuthorityOptions
+   * @property {boolean} [options.createAnother=true] Should create another authority for the wallet.
+   * @property {boolean} [options.startMiningTx=true] boolean to trigger start mining.
+   * @property {string} [options.pinCode] pin to decrypt xpriv information.
+   */
+
   /**
    * Prepare delegate authority transaction before mining
    *
    * @param {string} tokenUid UID of the token to delegate the authority
-   * @param {string} type Type of the authority to delegate 'mint' or 'melt'
+   * @param {string} type Type of the authority to delegate mint' or 'melt'
```

Comment:
Is it correct? I prefer without this change

Reply:
Changed